src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseSpanBasedStringConcat.Fixer.cs

```diff
@@ -0,0 +1,175 @@
+                return;
+
+            var operands = UseSpanBasedStringConcat.FlattenBinaryOperation(concatOperation);
+            //  Bail out if we don't have a long enough span-based string.Concat overload.
+            if (!symbols.TryGetRoscharConcatMethodWithArity(operands.Length, out IMethodSymbol? roscharConcatMethod))
+                return;
+
+            var codeAction = CodeAction.Create(
+                Resx.UseSpanBasedStringConcatTitle,
+                FixConcatOperationChain,
+                Resx.UseSpanBasedStringConcatTitle);
+            context.RegisterCodeFix(codeAction, diagnostic);
+
+            async Task<Document> FixConcatOperationChain(CancellationToken cancellationToken)
+            {
+                RoslynDebug.Assert(roscharConcatMethod is not null);
+
+                var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+                var generator = editor.Generator;
+
+                SyntaxNode stringTypeNameSyntax = generator.TypeExpressionForStaticMemberAccess(symbols.StringType);
+                SyntaxNode concatMemberAccessSyntax = generator.MemberAccessExpression(stringTypeNameSyntax, roscharConcatMethod.Name);
+
+                //  Save leading and trailing trivia so it can be attached to the outside of the string.Concat invocation node.
+                var leadingTrivia = operands.First().Syntax.GetLeadingTrivia();
+                var trailingTrivia = operands.Last().Syntax.GetTrailingTrivia();
+
+                var arguments = ImmutableArray.CreateBuilder<SyntaxNode>(operands.Length);
+                foreach (var operand in operands)
+                    arguments.Add(ConvertOperandToArgument(symbols, generator, operand));
+
+                //  Strip off leading and trailing trivia from first and last operand nodes, respectively, and
+                //  reattach it to the outside of the newly-created string.Concat invocation node.
+                arguments[0] = arguments[0].WithoutLeadingTrivia();
+                arguments[^1] = arguments[^1].WithoutTrailingTrivia();
+                SyntaxNode concatMethodInvocationSyntax = generator.InvocationExpression(concatMemberAccessSyntax, arguments.MoveToImmutable())
+                    .WithLeadingTrivia(leadingTrivia)
+                    .WithTrailingTrivia(trailingTrivia);
+
+                var newRoot = generator.ReplaceNode(root, concatExpressionSyntax, concatMethodInvocationSyntax);
+
+                //  Make sure 'System' namespace is imported.
+                if (!IsSystemNamespaceImported(generator.GetNamespaceImports(newRoot)))
+                {
+                    var systemNamespaceImport = generator.NamespaceImportDeclaration(nameof(System));
+                    newRoot = generator.AddNamespaceImports(newRoot, systemNamespaceImport);
+                }
```

Comment:
I think it's automatically imported in VB. The following (without `Imports System`) compiles on my machine:

```vb
Module Program
    Sub Main(args As String())
        Console.WriteLine("Hello World!")
    End Sub
End Module
```

If I added the extra `Imports System`, Visual Studio fades it out as unnecessary:

![image](https://user-images.githubusercontent.com/31348972/106639373-93d5f980-658d-11eb-8dcb-87a7446e181b.png)

Reply:
It's weird; when I build a project in Visual Studio, the `Imports System` statement isn't necessary. But when I don't import the namespace in the analyzer test, I get compiler error BC0456 "'AsSpan' is not a member of 'String'". Is the auto-importing of `System` a compiler switch in VB? I'll look into this some more.

Reply:
I figured it out. In VB, you can define "implicitly imported" namespaces for your project. Those namespaces will be considered to be imported for all files in the project. `System` is imported by default in visual studio, but apparently it's not imported by default in the test projects. I'm going to fix the analyzer so it detects implicitly-imported namespaces, and write tests that ensure `Imports System` is only added when it is not imported, either explicitly or implicitly.

Reply:
Fixed.